bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/CaddxCommunicator.java

```diff
@@ -0,0 +1,566 @@
+     * Adds this message before any others in the queue.
+     * Used by receiver to send ACKs.
+     *
+     * @param msg The message
+     */
+    public void transmitFirst(CaddxMessage msg) {
+        logger.trace("CaddxCommunicator.transmitFirst() Started");
+
+        messages.addFirst(msg);
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void run() {
+        logger.trace("CaddxCommunicator.run() Started");
+
+        int @Nullable [] expectedMessageNumbers = null;
+
+        @Nullable
+        CaddxMessage outgoingMessage = null;
+        boolean skipTransmit = false;
+
+        try {
+            // loop until the thread is interrupted, sending out messages
+            while (!Thread.currentThread().isInterrupted()) {
+                // Initialize the state
+                outgoingMessage = null;
+                expectedMessageNumbers = null;
+
+                if (!skipTransmit) {
+                    // send next outgoing message if we have one
+                    outgoingMessage = messages.poll();
+                    if (outgoingMessage != null) {
+                        logger.trace("CaddxCommunicator.run() Outgoing message: {}", outgoingMessage.getMessageType());
+                    }
+
+                    // Send the message
+                    if (outgoingMessage != null) {
+                        byte msg[] = outgoingMessage.getMessageFrameBytes(protocol);
+                        out.write(msg);
+                        out.flush();
+                    }
+
+                    // Log message
+                    if (outgoingMessage != null) {
+                        logger.info("->: {}", outgoingMessage.getName());
+                        logger.debug("{}", Util.buildCaddxMessageInBinaryString("->: ", outgoingMessage));
+                        // logger.debug("{}", Util.buildCaddxMessageInAsciiString("->: ", outgoingMessage));
+                    }
+                    if (outgoingMessage != null) {
+                        expectedMessageNumbers = outgoingMessage.getReplyMessageNumbers();
+                    }
+                } else {
+                    logger.trace("CaddxCommunicator.run() skipTransmit: true");
+                    skipTransmit = false;
+                }
+
+                // Check for an incoming message
+                CaddxMessage incomingMessage = null;
+                CaddxMessage throwAway = new CaddxMessage(new byte[] { 0x1d }, false);
+                try {
+                    incomingMessage = exchanger.exchange(throwAway, 3, TimeUnit.SECONDS);
+                } catch (TimeoutException e) {
+                    if (expectedMessageNumbers == null) { // Nothing expected, Nothing received we continue
+                        logger.trace("CaddxCommunicator.run(): Nothing expected, Nothing received we continue");
+                        continue;
+                    }
+                    logger.warn("CaddxCommunicator.run() TimeoutException caught.");
```

Comment:
I'm not sure this warrants logging as a warning

Reply:
@cpmeister  
I'm also not sure about this. 
Here is the case that the panel is not responding.
On the other hand if the panel is not responding the binding will flood the log file with warnings.

Reply:
I will convert it to debug